lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -159,10 +159,14 @@ SPIRVToLLVMDbgTran::transCompilationUnit(const SPIRVExtInst *DebugInst) {
   M->addModuleFlag(llvm::Module::Max, "Dwarf Version", DWARFVersion);
   SPIRVWord SourceLang =
       getConstantValueOrLiteral(Ops, LanguageIdx, DebugInst->getExtSetKind());
-  if (DebugInst->getExtSetKind() == SPIRVEIS_NonSemantic_Shader_DebugInfo_200)
+  if (DebugInst->getExtSetKind() == SPIRVEIS_NonSemantic_Shader_DebugInfo_200) {
     SourceLang = convertSPIRVSourceLangToDWARFNonSemanticDbgInfo(SourceLang);
-  else
+  } else {
+    if (!isSPIRVSourceLangValid(SourceLang))
```

Comment:
like this we call `convertSPIRVSourceLangToDWARF` anyway, shouldn't we not call it if isSPIRVSourceLangValid = false?

Reply:
> it seems that like this we call `convertSPIRVSourceLangToDWARF` anyway, shouldn't we not call it if isSPIRVSourceLangValid = false?

Right, then the call to `convertSPIRVSourceLangToDWARF` is really redundant. Thanks, fixed!